labels-config-service-impl/src/main/java/org/hypertrace/label/config/service/LabelsConfigServiceImpl.java

```diff
@@ -107,32 +103,23 @@ public void createLabel(
       CreateLabelRequest request, StreamObserver<CreateLabelResponse> responseObserver) {
     try {
       RequestContext requestContext = RequestContext.CURRENT.get();
-      Lock labelsLock = this.stripedLabelsLock.get(requestContext.getTenantId());
-      if (labelsLock.tryLock(WAIT_TIME.getSeconds(), TimeUnit.SECONDS)) {
-        try {
-          LabelData labelData = request.getData();
-          if (isDuplicateKey(requestContext, labelData.getKey())) {
-            // Creating a label with a name that clashes with one of system labels name
-            responseObserver.onError(new StatusRuntimeException(Status.ALREADY_EXISTS));
-            return;
-          }
-          Label label =
-              Label.newBuilder().setId(UUID.randomUUID().toString()).setData(labelData).build();
-          if (request.hasCreatedByApplicationRuleId()) {
-            label =
-                label.toBuilder()
-                    .setCreatedByApplicationRuleId(request.getCreatedByApplicationRuleId())
-                    .build();
-          }
-          Label createdLabel = labelStore.upsertObject(requestContext, label).getData();
-          responseObserver.onNext(CreateLabelResponse.newBuilder().setLabel(createdLabel).build());
-          responseObserver.onCompleted();
-        } finally {
-          labelsLock.unlock();
-        }
-      } else {
-        responseObserver.onError(new StatusRuntimeException(Status.ABORTED));
+      LabelData labelData = request.getData();
+      if (isDuplicateKey(requestContext, labelData.getKey())) {
```

Comment:
No longer strictly required, but I left these in place to get the better error message (vs getting the unique constraint violation)